requester/report.go

```diff
@@ -23,60 +23,53 @@ func (rep Report) String() string {
 	return string(b)
 }
 
-// collect pulls the records from Requester.Records as soon as
-// they are available and consumes them to build the report.
-// Returns the report when all the records have been collected.
-// Requester.collect will blocks until Requester.Records is empty.
-func (r *Requester) collect() Report {
-	rep := Report{
+// report generates and returns a Report from a previous Run.
+func (r *Requester) report() Report {
+	return Report{
```

Comment:
Should not happen but should we noop if `Run` was not called, ie `len(r.records) == 0 && r.numErr == 0` ?

Reply:
I wouldn't bother given it's private + it would return a zero Report anyway (except for the config).
We can consider making it a package func as a foolproof:
```go
func makeReport(cfg config.Config, records []Record, numErr int) Report
```

Reply:
Let's put that as a "nice to have" and move on then :)

Reply:
I'm fine making the change if we judge it preferable, it's a matter of seconds

Reply:
If it is quick, I think it does not eat bread :bread: to add it. I guess it makes more explicit that we have thought about this possibility and that it will not return a report in this case.